runtime/src/iree/hal/drivers/cuda/BUILD.bazel

```diff
@@ -59,6 +59,7 @@ iree_runtime_cc_library(
         "//runtime/src/iree/hal/utils:buffer_transfer",
         "//runtime/src/iree/hal/utils:collective_batch",
         "//runtime/src/iree/hal/utils:deferred_command_buffer",
+        "//runtime/src/iree/hal/utils:iree_hal_libmpi",
```

Comment:
`iree_hal_` seems redundant. How about utils:dylib_mpi?

Reply:
@benvanik PTAL and suggest a good naming.

Reply:
nit: change drivers/utils by hal/utils in the PR title.

Reply:
the main reason i prefixed it with iree_hal is that libmpi is too generic. TBH, i'm not a fan of dylib_mpi, so i'll wait for @benvanik to suggest a name that aligns with the project naming conventions. 

I did change the PR title.